### PR TITLE
Implement combined phonon band structure and dos plotting

### PR DESCRIPTION
**Pull request by utf**
Added a `--dos-file` option to `sumo-phonon-bandplot`, which expects a _total_dos.dat_ from Phonopy.
The colour is hard coded and the aspect ratio is not automatically corrected.
Also, only total dos is plotted rather than partial DOS.

Implemented by @ajjackson and @zccaayo.

### CODE REVIEW

File: sumo/cli/phonon_bandplot.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def phonon_bandplot(filename, poscar=None, prefix=None, directory=None,
                     symprec=0.01, mode='bradcrack', kpt_list=None,
                     eigenvectors=False, labels=None, height=6., width=6.,
                     ymin=None, ymax=None, image_format='pdf', dpi=400,
-                    plt=None, fonts=None):
+                    plt=None, fonts=None, dos_file=None):
```
Comment (utf) [inline sumo/cli/phonon_bandplot.py]: The option for combined DOS and electronic band structure plots is just `--dos`.

I think we should consider making them the same for consistency.

File: sumo/cli/phonon_bandplot.py [hunk 2]
```diff
@@ -114,6 +114,7 @@ def phonon_bandplot(filename, poscar=None, prefix=None, directory=None,
             A -> Z will be used instead.
         eigenvectors (:obj:`bool`, optional): Write the eigenvectors to the
             yaml file.
+        dos_file (str): Path to Phonopy total dos .dat file
```
Comment (utf) [inline sumo/cli/phonon_bandplot.py]: Very nit picky, but we should use the extended sphinx markup for optional function variables: e.g:

```
dos_file (:obj:`str`, optional): Etc.
```

File: sumo/plotting/phonon_bs_plotter.py [hunk 3]
```diff
@@ -35,8 +36,18 @@ def __init__(self, bs, imag_tol=-5e-2):
         PhononBSPlotter.__init__(self, bs)
         self.imag_tol = imag_tol
 
+    def _plot_phonon_dos(self, dos_file, ax=None):
+        if ax is None:
+            ax = plt.gca()
+        y, x = np.genfromtxt('total_dos.csv', delimiter=',', unpack=True)
```
Comment (utf) [inline sumo/plotting/phonon_bs_plotter.py]: The DOS data should be loaded in the `phonon_bandplot.py` script, with the data then passed through to the plotting object.

File: sumo/plotting/phonon_bs_plotter.py [hunk 4]
```diff
@@ -51,12 +62,33 @@ def get_plot(self, ymin=None, ymax=None, width=6., height=6., dpi=400,
...
+            dos_file (str): Path to total dos .csv data
+            dos_aspect (float): Width division for vertical DOS
 
         Returns:
             :obj:`matplotlib.pyplot`: The phonon band structure plot.
         """
-        plt = pretty_plot(width, height, dpi=dpi, plt=plt, fonts=fonts)
-        ax = plt.gca()
+
+        # if dos_plotter:
```
Comment (utf) [inline sumo/plotting/phonon_bs_plotter.py]: Do we still need these comments?

File: sumo/plotting/phonon_bs_plotter.py [hunk 5]
```diff
@@ -91,11 +125,16 @@ def _makeplot(self, ax, fig, data, ymin=None, ymax=None, height=6,
 
         ax.set_ylim(ymin, ymax)
         ax.set_xlim(0, data['distances'][-1][-1])
+        ax.axhline(0, color='k')
```
Comment (utf) [inline sumo/plotting/phonon_bs_plotter.py]: Might consider making this line grey and dashed?

Comment (ajjackson) [reply sumo/plotting/phonon_bs_plotter.py]: I'm no longer keen on the chunky dotted line I used in my thesis, it's too demanding of attention. Thin long dashes could be ok?

Comment (utf) [reply sumo/plotting/phonon_bs_plotter.py]: Yep, thin dashed lines should be fine.

The axis line width is in defined in `sumo.plotting._linewidth`. I think that seems like a sensible value to use for now. I guess we can make this inherited from the style sheet at a later date.

File: sumo/cli/phonon_bandplot.py (hunk 1 above)
Comment (ajjackson) [reply sumo/cli/phonon_bandplot.py]: done!

**Comment by ajjackson** [comment]
Have made the requested changes :-)

A few bonuses:
- The line colour is now drawn from the matplotlib style. The default looks fine for current versions of matplotlib and this should integrate with the improved style system (#13 ).
- The `--dos` argument can read a *total_dos.dat* from phonopy if available (no need to convert to CSV)
- If `--dos` is passed with no arguments a total dos will be generated using `--qmesh` mesh points.
- `--qmesh` defaults to (8, 8, 8)

**Comment by utf** [comment]
This looks great! Thanks Adam.

I might look into making a minimal style sheet before this is pulled into master, just to ensure that the colours are consistent with the rest of sumo.

**Comment by ajjackson** [comment]
That would be great! What do you think about having the default phonon band structure colour be different from the electronic band structure colours, even if they are all part of the main palette? I think it could aid clarity in reports that contain a mixture of both.

**Comment by ajjackson** [comment]
I'm trying to set this to read the colours from a matplotlib style, but at the moment the pretty_plot setup clobbers any style set at a higher level. The blue colour is actually the first colour from that series, so it _is_ a sumo colour, just not the one hard-coded for band plots. I think the best bet is to merge now and then get on with overhauling all the colouring.

In the meantime, I've changed it to read 'C2' from that palette so it's green :-)

**Comment by utf** [comment]
Great, looks good. Thanks Adam.

**Pull request merged**